File: 367.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        l, r = 1, num

        while(l <= r):
            m = (l + r)//2
            prod = m * m
            if(prod == num):
                return True
            elif(prod > num):
                r = m - 1
            else:
                l = m + 1
        return False

File: 367.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        int l = 1, r = num, m;
        unsigned long long prod;

        while(l <= r) {
            m = l + (r - l)/2;
            prod = (unsigned long long) m * m;
            if(prod == num) {
                return true;
            } else if (prod > num) {
                r = m - 1;
            } else {
                l = m + 1;
            }
        }
        return false;
    }
};